### PR TITLE
feat: add functionality to pass a custom name for the vpe gateway

### PR DESCRIPTION
**Pull request by MatthewLemmond**
### Description
Add a new variable to allow for passing a map containing keys which match the service name for which a VPE Gateway will be created corresponding to a value which is the name you want that gateway to have. Example and test for this functionality also created.

### Release required?
Identify the type of release. For information about the changes in a semantic versioning release, see [Release versioning](https://terraform-ibm-modules.github.io/documentation/#/versioning).

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content
Add the ability to customize the names of the Gateways being created by the module, in order to pass a name assign a map to `var.vpe_names` whose keys are the services which you want to have a custom gateway name for, and the corresponding values are the names you want to assign to those gateways.

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author. The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).
- Merge by using "Squash and merge".

### CODE REVIEW

File: examples/non-default/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# End-to-end example with non-default values
```
Comment (ocofaigh) [inline examples/non-default/README.md:1]: Lets be more specific in the name here.

**Review by shemau — CHANGES_REQUESTED** [review]
The implementation in the main module looks fine.

I would question the whole second example.
During CI tests both others_tests and pr_tests are going to run. Resulting in a second example test that is identical other the names on 3 out of 18 VPE gateways.

The new example code sets and passes the new variable and other_test.pr sends a value. I wonder if a better solution would be to put that code in the existing example and have it called from pr_test with the new map?
My thought is that the coverage is the same, both the old (default name, on 15 of 18 gateways) and the new (passed name, on 3 of 18 gateways) would be tested covering both code paths. This is achieved with only one VPC and ICD instance. Resource consumption is lower and the opportunity to fail is lower (VPC tear down has been a bit flaky at times).

**Review by shemau — APPROVED** [review]
LGTM

**Pull request merged**